Sources/SwiftDiscord/DiscordSnowflakeID.swift

```diff
@@ -108,12 +108,32 @@ extension Snowflake {
 
 // MARK: Snowflake Conformances
 
-/// Snowflake conformance to JSONRepresentable
-extension Snowflake : JSONRepresentable {
-    func jsonValue() -> JSONRepresentable {
-        // Snowflakes should be put into JSON as Strings
-        return self.description
+extension Snowflake: Codable {
+    /// Set this key to true on an encoder to encode Snowflakes as UInt64s instead of Strings.
+    /// This will save space in binary encoders like binary plists, but will cause encoded JSON to be
+    /// incompatible with Discord. Since JSON isn't a binary encoding, it won't save much space there anyways.
+    public static let encodeAsUInt64 = CodingUserInfoKey(rawValue: "snowflakeAsUInt64")!
+    public init(from decoder: Decoder) throws {
+        do {
+            let intForm = try UInt64(from: decoder)
+            self = Snowflake(intForm)
+        }
+        catch _ {
+            let stringForm = try String(from: decoder)
```

Comment:
If you only use a variable once, you can toss it and move it where it's used. Some exceptions are for expressions that getting busy or when you're at a line length limit (I use 120).